static/src/javascripts/bootstraps/enhanced/notifications.js

```diff
@@ -0,0 +1,180 @@
+    template,
+    mediator,
+    robust,
+    svgs,
+    userPrefs,
+    subscribeTemplate,
+    some,
+    uniq,
+    without
+) {
+    var modules, reg, sub,
+        isSubscribed = false;
+
+    modules = {
+
+        getPushSubscription: function () {
+
+            navigator.serviceWorker.ready.then(function (serviceWorkerRegistration) {
+                reg = serviceWorkerRegistration;
+                serviceWorkerRegistration.pushManager.getSubscription().then(function (pushSubscription) {
+
+                    if (pushSubscription) {
+                        sub = pushSubscription;
+                        modules.configureSubscribeTemplate();
+                    } else {
+                        reg.pushManager.subscribe({userVisibleOnly: true}).then(
+                            function (pushSubscription) {
+                                sub = pushSubscription;
+                                modules.configureSubscribeTemplate();
+                            }
+                        );
+                    }
+                });
+            });
+        },
+
+        configureSubscribeTemplate: function () {
+            var subscribed = modules.checkSubscriptions(),
+                hasNoSubscriptions = modules.subscriptionsEmpty(),
+                src = template(subscribeTemplate, {
+                    className: hasNoSubscriptions ? '' : 'notifications-subscribe-link--has-subscriptions',
+                    text: subscribed ? 'Following story' : 'Follow story',
+                    imgMobile: svgs('notificationsExplainerMobile', ['mobile-only', 'notification-explainer']),
+                    imgDesktop: svgs('notificationsExplainerDesktop', ['hide-on-mobile', 'notification-explainer'])
+                });
+
+            fastdom.write(function () {
+                $('.js-liveblog-body').prepend(src);
+                mediator.emit('page:notifications:ready');
+            });
+
+            bean.on(document.body, 'click', '.js-notifications-subscribe-link', function () {
+                modules.buttonHandler();
+            });
+        },
+
+        setSubscriptionStatus: function (subscribed) {
+            var subscribeButton = $('.js-notifications-subscribe-link'),
+                subscribeLink = $('.notifications-subscribe-link--follow');
+            isSubscribed = subscribed;
+            if (subscribed) {
+                subscribeLink.addClass('notifications-subscribe-link--has-subscriptions');
+            }
+            subscribeButton[0].textContent = subscribed ?  'Following story' : 'Follow story';
+        },
+
+        buttonHandler: function () {
+            if (isSubscribed) {
+                modules.stopFollowingThis();
+            } else {
+                modules.subscribe();
+            }
+        },
+
+        subscribe: function () {
+            if (modules.subscriptionsEmpty()) {
+                reg.pushManager.subscribe({userVisibleOnly: true}).then(function (pushSubscription) {
```

Comment:
@OliverJAsh . Thanks! That makes sense I'll give that I try and give it a test ( ill shout if i get stuck .. )